crates/full-node/sov-sequencer/src/preferred/cache_warm_up_executor.rs

```diff
@@ -8,14 +9,34 @@ use sov_modules_api::Spec;
 use sov_modules_api::StateCheckpoint;
 use sov_modules_api::StateUpdateInfo;
 use sov_modules_api::Storage;
+use sov_modules_api::TxChangeSet;
 use sov_modules_api::{FullyBakedTx, Runtime};
 use std::collections::BTreeMap;
+use tokio::sync::oneshot;
 use tokio::task::JoinHandle;
 
 // We have several work-stealing executor workers for a single main worker, so we don't expect the channel to become full.
 // Even if it does, the sender uses a non-blocking method, meaning a few updates may simply be skipped.
 const TX_CHANNEL_SIZE: usize = 16;
 
+/// A transaction with a way to retrieve the corresponding state changes it may produce.
+pub struct FullyBakedTxWithMaybeChangeSet {
```

Comment:
nit: This also looks like a misnomer to me since (IIUC) we're receiving the *read* set not the write set